skimage/graph/_graph.py

```diff
@@ -72,8 +75,9 @@ def pixel_graph(image, *, mask=None, edge_function=None, connectivity=1, spacing
     """
     if image.dtype == bool and mask is None:
         mask = image
-    if mask is None and edge_function is None:
+    if mask is None:
         mask = np.ones_like(image, dtype=bool)
+    if edge_function is None:
         edge_function = _weighted_abs_diff
```

Comment:
With this change, `skimage/graph/tests/test_pixel_graph.py::test_small_graph` fails at the `_weighted_abs_diff` call, while, previously (i.e., current `main`), `edge_function` would remain `None` and we would later enter this conditional:
https://github.com/scikit-image/scikit-image/blob/28454b08ab3ae8e995a9681073b817cdbe60a8fd/skimage/graph/_graph.py#L121-L122
Note that, in this test, `mask` doesn't remain `None` because of
https://github.com/scikit-image/scikit-image/blob/28454b08ab3ae8e995a9681073b817cdbe60a8fd/skimage/graph/_graph.py#L73-L74
right above. You need hierarchy in the above conditionals.